### ISSUE TITLE
[import/default] - false negative for react-test-renderer

### ISSUE THREAD

**Issue by Maxim-Mazurok** [import-js/eslint-plugin-import#2183]
I'm doing this:
```
import reactTestRenderer from 'react-test-renderer';
```

Getting this:
```
No default export found in imported module "react-test-renderer". eslint (import/default)
```

Using these types: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-test-renderer/index.d.ts

And I have `esModuleInterop` in my tsconfig, so this should not throw error (thanks to https://github.com/import-js/eslint-plugin-import/pull/1689)

**Comment by alasdairhurst** [comment]
This may be related to this, what seems like a regression in the latest version of elsint-plugin-import:

We're using mingo@2.5.3

```
const Mingo = require('mingo');
```

Suddenly started resulting in the error:
```
Unable to resolve path to module 'mingo' import/no-unresolved
```

Not fully versed with how typescript imports work, but Mingo also uses it. My module does not.
https://github.com/kofrasa/mingo/tree/2.5.3

note we're also not using the "import" syntax here, but require.  Maybe plugin-import is mixing up "main" and "module" in package.json?

```
"main": "./dist/mingo.js",
 "module": "./lib/index.js",
```

the file at "module" is not available in this case.
![image](https://user-images.githubusercontent.com/8441810/128748028-9475cbec-9246-45c3-9678-f04e02034f21.png)

**Comment by Maxim-Mazurok** [comment]
@alasdairhurst I didn't put a lot of effort into figuring out if this is regression or not, your issue is probably different... Probably you should open a new one.

**Comment by ljharb** [comment]
@alasdairhurst i don't think that is related; please do file a new issue.
